src/interpreter/mod.rs

```diff
@@ -844,6 +844,47 @@ where
                         None => return Some(Err(Error::UnexpectedStackEnd)),
                     }
                 }
+                Terminal::MultiA(k, ref subs) => {
+                    if node_state.n_evaluated == subs.len() {
+                        if node_state.n_satisfied == k {
+                            self.stack.push(stack::Element::Satisfied);
+                        } else {
+                            self.stack.push(stack::Element::Dissatisfied);
+                        }
+                    } else {
+                        // evaluate each key with as a pk
+                        // note that evaluate_pk will error on non-empty incorrect sigs
+                        // push 1 on satisfied sigs and push 0 on empty sigs
+                        match self
+                            .stack
+                            .evaluate_pk(&mut self.verify_sig, &subs[node_state.n_evaluated])
```

Comment:
wouldn't `subs[node_state.n_evaluated]` panic if the number of the provided signatures is less than k?

Reply:
`n_evaluated` denotes the numbers of sub-expressions evaluated, not the number of sigs. The number of expressions evaluated must be less than subs.len() because we add +1 whenever we evaluate one sub-expression. 

Inside `evalute_pk`, there is a check that checks if we have one sig for a given pk. This will correctly error if given sigs is less than the number of keys.

Reply:
At some distant future a test for that will be nice to have :)

Reply:
Will add in this one. Should be a couple mins of work